agent/provider/src/payments/payments.rs

```diff
@@ -209,7 +211,7 @@ async fn send_debit_note(
         activity_id: debit_note_info.activity_id.clone(),
         total_amount_due: cost_info.cost,
         usage_counter_vector: Some(json!(cost_info.usage)),
-        payment_due_date: None,
+        payment_due_date: debit_note_info.payment_timeout.map(|dur| Utc::now() + dur),
```

Comment:
Spec says, that DebitNotes should be sent relative to Agreement timestamp `approved_ts`. This complicates things, but we need to have absolute reference point. Current Provider implementation isn't caring about this, so it requires changes